src/LandIce/evaluators/LandIce_Gather2DField_Def.hpp

```diff
@@ -357,8 +363,14 @@ evaluateFields(typename Traits::EvalData workset)
   bool g_xx_is_active = !workset.hessianWorkset.hess_vec_prod_g_xx.is_null();
   bool g_xp_is_active = !workset.hessianWorkset.hess_vec_prod_g_xp.is_null();
   bool g_px_is_active = !workset.hessianWorkset.hess_vec_prod_g_px.is_null();
+  bool f_xx_is_active = !workset.hessianWorkset.hess_vec_prod_f_xx.is_null();
+  bool f_xp_is_active = !workset.hessianWorkset.hess_vec_prod_f_xp.is_null();
+  bool f_px_is_active = !workset.hessianWorkset.hess_vec_prod_f_px.is_null();
+
+  const bool is_active = g_xx_is_active || g_xp_is_active || f_xx_is_active || f_xp_is_active;
```

Comment:
Can you add a comment explaining what `is_active` and `is_direction_active` are?